src/sqlancer/postgres/gen/PostgresWindowFunctionGenerator.java

```diff
@@ -0,0 +1,138 @@
+        throw new AssertionError("Utility class should not be instantiated");
+    }
+
+    public static PostgresWindowFunction generateWindowFunction(PostgresGlobalState globalState,
+            List<PostgresExpression> availableExpr) {
+
+        String functionName = selectRandomWindowFunction();
+        List<PostgresExpression> arguments = generateFunctionArguments(functionName, globalState, availableExpr);
+        WindowSpecification windowSpec = generateWindowSpecification(globalState, availableExpr);
+        PostgresDataType returnType = determineReturnType(functionName);
+
+        return new PostgresWindowFunction(functionName, arguments, windowSpec, returnType);
+    }
+
+    private static String selectRandomWindowFunction() {
+        return Randomly.fromList(WINDOW_FUNCTIONS);
+    }
+
+    private static List<PostgresExpression> generateFunctionArguments(String functionName,
+            PostgresGlobalState globalState, List<PostgresExpression> availableExpr) {
+        List<PostgresExpression> arguments = new ArrayList<>();
+
+        switch (functionName) {
+        case "ntile":
+            arguments
+                    .add(PostgresExpressionGenerator.generateConstant(globalState.getRandomly(), PostgresDataType.INT));
+            break;
+        case "lag":
+        case "lead":
+        case "nth_value":
+            arguments.add(Randomly.fromList(availableExpr));
+            if (Randomly.getBoolean()) {
+                arguments.add(
+                        PostgresExpressionGenerator.generateConstant(globalState.getRandomly(), PostgresDataType.INT));
+            }
+            break;
+        case "first_value":
+        case "last_value":
+            arguments.add(Randomly.fromList(availableExpr));
+            break;
+        default:
+            // No arguments needed for other window functions
+            break;
+        }
+
+        return arguments;
+    }
+
+    private static WindowSpecification generateWindowSpecification(PostgresGlobalState globalState,
+            List<PostgresExpression> availableExpr) {
+        List<PostgresExpression> partitionBy = generatePartitionByClause(availableExpr);
+        List<PostgresOrderByTerm> orderBy = generateOrderByClause(availableExpr);
+        WindowFrame frame = generateWindowFrame(globalState);
+
+        return new WindowSpecification(partitionBy, orderBy, frame);
+    }
+
+    private static List<PostgresExpression> generatePartitionByClause(List<PostgresExpression> availableExpr) {
+        List<PostgresExpression> partitionBy = new ArrayList<>();
+        if (Randomly.getBooleanWithRatherLowProbability()) {
+            int count = Randomly.smallNumber();
+            for (int i = 0; i < count; i++) {
+                partitionBy.add(Randomly.fromList(availableExpr));
+            }
+        }
+        return partitionBy;
+    }
+
+    private static List<PostgresOrderByTerm> generateOrderByClause(List<PostgresExpression> availableExpr) {
```

Comment:
We already have a method in the `PostgresExpressionGenerator`, so I think we should use that one.